rust/src/node_manager/lock_cache/shuttle_tests.rs

```diff
@@ -270,6 +275,105 @@ impl PermutationTestCase {
     }
 }
 
+/// Tests all permutations of operations and a small set of node IDs on the lock cache using
+/// shuttle. The test is repeated for different cache sizes to stress cache contention scenarios.
+/// The idea is to find cases where the lock cache guarantees are violated, i.e. an operation
+/// returns a reference to a non-existing node or an unexpected error occurs.
+/// The permutation generates known invalid operation sequences that are expected to fail, and
+/// these are reported but do not cause the test to fail.
+/// To facilitate debugging, the test case that caused a failure is serialized to a file, and
+/// can be reloaded by setting the `SHUTTLE_SERIALIZED_CASE` environment variable.
+#[test]
+fn shuttle__operation_permutations() {
+    #[cfg(not(feature = "shuttle"))]
+    return;
+    #[allow(unreachable_code)]
```

Comment:
This might read a little nicer, but no string opinions.
```suggestion
    if cfg!(not(feature = "shuttle")) {
        return;
    }
```